src/ansiblelint/rules/complexity.py

```diff
@@ -42,6 +45,32 @@ def matchplay(self, file: Lintable, data: dict[str, Any]) -> list[MatchError]:
             )
         return results
 
+    def matchtask(self, task: Task, file: Lintable | None = None) -> list[MatchError]:
+        """Check if the task is a block and count the number of items inside it."""
+        results: list[MatchError] = []
+
+        if task.get("action") == "block":
```

Comment:
@Ruchip16 This condition will not return true as `task.get("action")` returns a dict instead of a string